dali/base/sysinfologger.cpp

```diff
@@ -0,0 +1,891 @@
+    {
+        return hiddenOnly;
+    }
+    virtual bool queryVisibleOnly() const override
+    {
+        return visibleOnly;
+    }
+    virtual unsigned __int64 queryMatchTimeStamp() const override
+    {
+        return matchTimeStamp;
+    }
+    virtual unsigned queryStartYear() const override
+    {
+        return matchStartYear;
+    }
+    virtual unsigned queryStartMonth() const override
+    {
+        return matchStartMonth;
+    }
+    virtual unsigned queryStartDay() const override
+    {
+        return matchStartDay;
+    }
+    virtual unsigned queryEndYear() const override
+    {
+        return matchEndYear;
+    }
+    virtual unsigned queryEndMonth() const override
+    {
+        return matchEndMonth;
+    }
+    virtual unsigned queryEndDay() const override
+    {
+        return matchEndDay;
+    }
+    virtual unsigned queryMatchYear() const override
+    {
+        return (matchStartYear==matchEndYear ? matchStartYear : 0);
+    }
+    virtual unsigned queryMatchMonth() const override
+    {
+        return (matchStartMonth==matchEndMonth ? matchStartMonth : 0);
+    }
+    virtual unsigned queryMatchDay() const override
+    {
+        return (matchStartDay==matchEndDay ? matchStartDay : 0);
+    }
+    virtual const char * queryMatchSource() const override
+    {
+        return matchSource.str();
+    }
+    virtual LogMsgCode queryMatchCode() const override
+    {
+        return matchCode;
+    }
+    virtual LogMsgAudience queryMatchAudience() const override
+    {
+        return matchAudience;
+    }
+    virtual LogMsgClass queryMatchClass() const override
+    {
+        return matchClass;
+    };
+    virtual unsigned __int64 queryMatchMsgId() const override
+    {
+        return matchMsgId;
+    };
+
+};
+
+ISysInfoLoggerMsgFilter * createSysInfoLoggerMsgFilter()
+{
+    return new CSysInfoLoggerMsgFilter();
+}
+
+class CSysInfoLoggerMsgIterator : public CInterface, implements ISysInfoLoggerMsgIterator
+{
+    Linked<ISysInfoLoggerMsgFilter> filter;
+    Owned<IRemoteConnection> conn = nullptr;
+    bool updateable = false;
+    Owned<IPropertyTree> root = nullptr;
+    Owned<IPropertyTreeIterator> msgIter = nullptr;
+    CSysInfoLoggerMsg infoMsg;
+
+    bool ensureMatch()
+    {
+        if (filter->hasDateRange())
+        {
+            for (; msgIter && msgIter->isValid(); msgIter->next())
+            {
+                unsigned __int64 ts = msgIter->query().getPropInt64(ATTR_TIMESTAMP, 0);
+                if(!ts || !filter->isInDateRange(ts))
+                    continue;
+                break;
+            }
+        }
+        return msgIter->isValid();
+    }
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    CSysInfoLoggerMsgIterator(ISysInfoLoggerMsgFilter * _filter, bool _updateable=false) : filter(_filter), updateable(_updateable)
+    {
+    }
+    CSysInfoLoggerMsg & queryInfoLoggerMsg()
+    {
+        return infoMsg.set(msgIter->get(), *(root.get()), updateable);
```

Comment:
linking her is a bit confusing.
1st parameter links , 2nd doesn't (but both unfortunately called get()).
Then set owns 1st and double links 2nd (see other comment).

But anyway, these items cannot be valid beyond the life of the iterator, since infoMsg will be destroyed, so I don't think any value in linking either (NB: they will remain valid as long as 'conn' does anyway).